src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -389,6 +389,60 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         return _terminal->SendCharEvent(ch, scanCode, modifiers);
     }
 
+    bool ControlCore::TryMarkModeKeybinding(const WORD vkey,
+                                            const ::Microsoft::Terminal::Core::ControlKeyStates modifiers)
+    {
+        if (HasSelection() &&

```

Comment:
Methinks

```c++
            HasSelection() &&
            !KeyEvent::IsModifierKey(vkey) &&
            vkey != VK_SNAPSHOT &&
```

should probably also be pulled into a helper, so that we don't only alter one of `TryMarkModeKeybinding` and `TrySendKeyEvent`